src/redemptions/redemption.service.ts

```diff
@@ -167,8 +167,7 @@ export class RedemptionService {
     });
   }
 
-  async canAttempt(
-    redemption: Redemption | null,
+  async isEligible(
```

Comment:
nit: it's a bit odd that the method name is `isEligible` but is returning an error or null. It might be easier to follow if we return a boolean along with an error reason, similar to how we do in the node.